insitu_plugins/src/Stats/Stats.cpp

```diff
@@ -0,0 +1,176 @@
+#include <Stats/Stats.hpp>
+#include <Stats/Stats_dialog.hpp>
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <sstream>
+#include <sys/statvfs.h>
+#include <thread>
+
+using namespace std::chrono_literals;
+
+namespace insitu_plugins
+{
+/*
+    Filter Implementation
+*/
+Stats::Stats(void)
+{
+    // TODO instantiation code
+}
+
+void Stats::filterInit(void)
+{
+    settingsDialog = new StatsDialog(this);
+    setSize(QSize(200, 120));
+}
+
+void Stats::onDelete(void)
+{
+}
+
+void Stats::drawtorect(cv::Mat& mat, cv::Rect target, const std::string& str,
+                       int face, int thickness, cv::Scalar color)
+{
+    cv::Size rect = cv::getTextSize(str, face, 1.0, thickness, 0);
+    double scalex = (double)target.width / (double)rect.width;
+    double scaley = (double)target.height / (double)rect.height;
+    double scale = std::min(scalex, scaley);
+    int marginx =
+        scale == scalex ?
+            0 :
+            (int)((double)target.width * (scalex - scale) / scalex * 0.5);
+    int marginy =
+        scale == scaley ?
+            0 :
+            (int)((double)target.height * (scaley - scale) / scaley * 0.5);
+    cv::putText(
+        mat, str,
+        cv::Point(target.x + marginx, target.y + target.height - marginy), face,
+        scale, color, thickness, cv::LINE_AA, false);
+}
+
+const cv::Mat Stats::apply(void)
+{
+    std::this_thread::sleep_for(1s);
```

Comment:
The `apply` function is called every image (30 hz) right? Are there ramifications for sleeping here and not returning quickly?

Reply:
Apply runs asynchronously from the main thread so sleeping/waiting within the filter does not affect the main GUI, your specific filter just won't update as quickly.